packages/odyssey-react-mui/src/labs/Drawer.tsx

```diff
@@ -0,0 +1,258 @@
+}>`
+  overflow-y: auto;
+`;
+
+const DrawerContent = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<DrawerStyleProps>`
+  padding: ${({ showDividers, odysseyDesignTokens }) =>
+    showDividers
+      ? `${odysseyDesignTokens.Spacing5}`
+      : `0 ${odysseyDesignTokens.Spacing5}`};
+`;
+
+const DrawerFooter = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<DrawerStyleProps>`
+  position: sticky;
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  bottom: 0;
+  background-color: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.HueNeutralWhite};
+  padding: ${({ odysseyDesignTokens }) => odysseyDesignTokens.Spacing4};
+  align-content: center;
+  border-top: ${({ showDividers, odysseyDesignTokens }) =>
+    showDividers ? `1px solid ${odysseyDesignTokens.HueNeutral200}` : "none"};
+`;
+
+const Drawer = ({
+  ariaLabel,
+  children,
+  isOpen,
+  onClose,
+  primaryCallToActionComponent,
+  secondaryCallToActionComponent,
+  showDividers = false,
+  tertiaryCallToActionComponent,
+  testId,
+  title,
+  translate,
+  variant = "temporary",
+}: DrawerProps) => {
+  const [isContentScrollable, setIsContentScrollable] = useState(false);
+  const drawerContentRef = useRef<HTMLDivElement>(null);
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  //If RTL is set in the theme, align the drawer on the left side of the screen, uses right by default.
+  const { i18n } = useTranslation();
+  const anchorDirection = i18n.dir() === "rtl" ? "left" : "right";
+
+  useEffect(() => {
+    let frameId: number;
+
+    const handleContentScroll = () => {
+      const drawerContentElement = drawerContentRef.current;
+      if (drawerContentElement) {
+        cancelAnimationFrame(frameId);
+        setIsContentScrollable(
+          drawerContentElement.scrollHeight > drawerContentElement.clientHeight,
+        );
+      }
+      frameId = requestAnimationFrame(handleContentScroll);
+    };
+
+    if (isOpen) {
+      frameId = requestAnimationFrame(handleContentScroll);
+    }
+
+    return () => {
+      cancelAnimationFrame(frameId);
+    };
+  }, [isOpen]);
+
+  const dividersVisible = useMemo(() => {
+    return showDividers || isContentScrollable;
+  }, [showDividers, isContentScrollable]);
+
+  const hasFooter = useMemo(
+    () =>
+      primaryCallToActionComponent ||
+      secondaryCallToActionComponent ||
+      tertiaryCallToActionComponent,
+    [
+      primaryCallToActionComponent,
+      secondaryCallToActionComponent,
+      tertiaryCallToActionComponent,
+    ],
+  );
+
+  return (
+    <MuiDrawer
+      data-se={testId}
+      anchor={anchorDirection}
+      open={isOpen}
+      onClose={onClose}
+      variant={variant}
+      sx={{
+        //Overrides defualt MUI inline style
+        ...(variant === "persistent" && {
+          "& .MuiDrawer-paper": {
+            transition: "none",
+          },
+        }),
+      }}
```

Comment:
Not against this, but I thought our convention was to not do this? Would it make sense to have a separate, styled `MuiDrawer` that we can use if the `variant === "persistent"`?
for example, something like::
```
if (variant === "persistent") {
  return (
    <AltMuiDrawer ... />
  )
}
return <MuiDrawer ... />
```

Reply:
@chrispulsinelli-okta I know we generally frown on using `sx` but in this instance, the `transition` is applied dynamically by MUI to the child element ' MuiDrawer-paper` for a slide transition. There does not appear to be a way to disable this so I think the only option is either to use `!important` in a style override or add a `sx` dynamically.